### DISCUSSION TITLE
How to add onClick to Disclosure.Button?

### DISCUSSION THREAD

**Discussion by btahir** [tailwindlabs/headlessui#640] [Help]
Hello - 

It seems adding onClick to Disclosure.Button does not do anything. I know I can achieve this by adding a button nested within Disclosure but was wondering if there is a more elegant way to do this? 

Thanks!

**Comment by btahir** [comment]
I think I found the answer in another issue. It seems we use onPointerUp instead of onClick. Not sure why both are not supported but as long as it works. :)

**Reply by samtsai** [reply]
You can also add an `onClick` like this:

```
<Disclosure.Button as={Fragment}>
        <button onClick={handler}>What languages do you support?</button>
      </Disclosure.Button>
```
